flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskExecutorRunner.java

```diff
@@ -97,22 +91,7 @@ private static void run(String[] args) {
 			final String currDir = ENV.get(Environment.PWD.key());
 			LOG.info("Current working Directory: {}", currDir);
 
-      // Some dynamic properties for task manager are added to args, such as managed memory size.
-			final CommandLineParser<ClusterConfiguration> commandLineParser =
-				new CommandLineParser<>(new ClusterConfigurationParserFactory());
-
-			ClusterConfiguration clusterConfiguration = null;
-			try {
-				clusterConfiguration = commandLineParser.parse(args);
-			} catch (FlinkParseException e) {
-				LOG.error("Could not parse command line arguments {}.", args, e);
-				commandLineParser.printHelp(YarnTaskExecutorRunner.class.getSimpleName());
-				System.exit(1);
```

Comment:
It seems we changed the behavior here. The `System.exit(1)` is replaced by throwing exception after reusing `TaskManagerRunner.loadConfiguration` below. 
Another option is catching `TaskManagerRunner.loadConfiguration` to exit as previous way.

Reply:
Or we can further reuse `TaskManagerRunner#runTaskManagerSecurely` instead?

Reply:
I think we should not `System.exit(1)` here. We should use `INIT_ERROR_EXIT_CODE` instead. 

We just can not use `TaskManagerRunner#runTaskManagerSecurely` for `YarnTaskExecutorRunner`. Because the configuration needs to be updated and then use the updated configuration to start task executor. It is the unique logics of Yarn.

Reply:
Sorry I missed the below catch clause for catching this exception and exiting finally.
If so it makes sense for me.